app/controllers/charge_controller.js

```diff
@@ -198,9 +203,17 @@ module.exports = {
     const charge = normalise.charge(req.chargeData, req.chargeId)
     const issuerUrl = _.get(charge, 'auth3dsData.issuerUrl')
     const paRequest = _.get(charge, 'auth3dsData.paRequest')
+    const md = _.get(charge, 'auth3dsData.md')
     const htmlOut = _.get(charge, 'auth3dsData.htmlOut')
 
-    if (issuerUrl && paRequest) {
+    if (issuerUrl && paRequest && md) {
```

Comment:
This could be simplified to be something like this;
```
if (issuerUrl && paRequest) {
      let data = {
        issuerUrl: issuerUrl,
        paRequest: paRequest,
        threeDSReturnUrl: `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', {chargeId: charge.id})}`
      }
      if (md) {
        data.md = md
      }
      views.display(res, AUTH_3DS_REQUIRED_OUT_VIEW, data)
    }
```